sklearn/decomposition/_kernel_pca.py

```diff
@@ -363,6 +365,14 @@ def _fit_transform_in_place(self, K):
                 random_state=self.random_state,
                 selection="module",
             )
+        elif eigen_solver == "randomized_value":
+            self.eigenvalues_, self.eigenvectors_ = _randomized_eigsh(
```

Comment:
Maybe we could add comments to this branch and the previous branch of the if so that it is easier to remember the difference while maintaining the code

`# Use selection='module' for SVD decomposition (wrong if non-PSD precomputed kernel)

and

```suggestion
            # Use selection='value' for direct eigenvalue decomposition (robust to non-PSD)
            self.eigenvalues_, self.eigenvectors_ = _randomized_eigsh(
```